### PR TITLE
Send Feedback event subtypes

### PR DESCRIPTION
**Pull request by avi-c**
Fix for #2618 

Add JSON keys for Feedback event subtypes. Update the feedback events with the subtypes before sending them back to the telemetry endpoint.

### CODE REVIEW

File: MapboxCoreNavigation/EventDetails.swift [hunk 1]
```diff
@@ -352,7 +352,9 @@ enum EventDetailsError: Error {
 
 extension EventDetails {
     func asDictionary() throws -> [String: Any] {
-        let data = try JSONEncoder().encode(self)
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
```
Comment (1ec5) [inline MapboxCoreNavigation/EventDetails.swift:356]: I think I was mistaken in https://github.com/mapbox/mapbox-navigation-ios/issues/2618#issue-699592630: so far, we’ve only identified that the values of this one key are in the wrong case, but the keys aren’t necessarily in the wrong case. After all, the general feedback event code hasn’t changed for quite a long time. Looking at the backend’s event schema, I think camel case is actually correct for the keys.

File: MapboxCoreNavigation/Feedback.swift [hunk 2]
```diff
@@ -26,22 +26,85 @@ public enum FeedbackType: CustomStringConvertible {
 
     /// Indicates a problem with positioning the user
     case positioning(subtype: PositioningSubtype?)
-    
+
+    /*
+      case streetPermanentlyBlockedOff
+      case roadMissingFromMap
+      case other
```
Comment (1ec5) [inline MapboxCoreNavigation/Feedback.swift:33]: What’s this comment for?

File: MapboxCoreNavigation/Feedback.swift [hunk 3]
```diff
@@ -26,22 +26,85 @@ public enum FeedbackType: CustomStringConvertible {
...
     public var description: String {
         switch self {
         case .general:
             return "general"
-        case .incorrectVisual(_):
+        case .incorrectVisual(subtype: .none):
             return "incorrect_visual"
-        case .confusingAudio(_):
+        case .incorrectVisual(subtype: .turnIconIncorrect):
+            return "turn_icon_incorrect"
```
Comment (1ec5) [inline MapboxCoreNavigation/Feedback.swift:42]: This replaces the type with the subtype in the event. Let’s verify that that’s what the server expects and what the Android navigation SDK is sending. At a glance, I’m unsure how this code would be able to send multiple subtypes for the same event if the user selects multiple rows in the feedback detail view controller – would it be multiple events?

File: MapboxCoreNavigation/Feedback.swift (hunk 2 above)
Comment (avi-c) [reply MapboxCoreNavigation/Feedback.swift:33]: I'll remove it. I was dropping in the enum as a reference as I built the case statement

File: MapboxCoreNavigation/EventDetails.swift (hunk 1 above)
Comment (avi-c) [reply MapboxCoreNavigation/EventDetails.swift:356]: Let's remove this then until it's clear it's needed.

File: MapboxCoreNavigation/Feedback.swift (hunk 3 above)
Comment (avi-c) [reply MapboxCoreNavigation/Feedback.swift:42]: As I understand it, the way it sends multiple events is that for each `item` selected in the UI, it creates a new FeedbackEvent (by getting a new UUID for one), then updates that latest event with the actual event details.

Comment (1ec5) [reply MapboxCoreNavigation/Feedback.swift:42]: Based on a cursory reading of the events schema, the SDK needs to send a single feedback event that contains both the type and any subtypes (in the `feedbackSubType` property).

**Comment by 1ec5** [comment]
Cherry-picked into the release-v1.0 branch as part of 18b89363fd8d36210372354383b034f1e65b6ddd.

**Pull request merged**